cloud/src/recycler/s3_accessor.cpp

```diff
@@ -263,8 +354,30 @@ int S3Accessor::init() {
         // Therefore, you only need to add a policy to check if the response code is 429 and if the retry count meets the condition, it can record the retry count.
         options.PerRetryPolicies.emplace_back(
                 std::make_unique<AzureRetryRecordPolicy>(config::max_s3_client_retry));
+
         auto container_client = std::make_shared<Azure::Storage::Blobs::BlobContainerClient>(
                 uri_, cred, std::move(options));
+
+        try {
+            // Test Azure connection
+            auto connection_result = testAzureConnection(container_client, conf_.bucket);
+            switch (connection_result) {
+            case ConnectionTestResult::SUCCESS:
+                break;
+            case ConnectionTestResult::AUTH_ERROR:
+                LOG(WARNING) << ("Azure S3 authentication failed");
+                return -1;
+            case ConnectionTestResult::TIMEOUT_ERROR:
+                LOG(WARNING) << ("Azure S3 connection timed out");
+                return -1;
+            default:
+                LOG(WARNING) << ("Azure S3 connection failed");
+                return -1;
+            }
+        } catch (const std::exception& e) {
```

Comment:
`testAzureConnection()` already catch all exceptions why do we need this?
and the switch case can be hidden in `testAzureConnection()` too.